### ISSUE TITLE
Qt Learning Spike: Overlays built-into UI

### ISSUE THREAD

**Issue by ninavizz** [freedomofpress/securedrop-client#1372]
## Problem
Traditionally, "overlays" in client apps are not dialogs—rather, they provide they're modal UIs that provide the visual illusion of sitting atop the regular client chrome, yet they're just high z-axis dealie-os (when the designer tries to get technical—bear with me, heh). QT has z-axis/layering issues... which may be the version of QT we're limited to by PyQT, or is also an issue on the latest version of QT more plausible to work with after updating the packaging wrapper. (I think is how Allie explained it—will just stop and bat my eyelashes and smile and be charming at this point). 

## Solution
Learning spike to learn how to mimic visual "layering" of UI panels atop other UI objects in QT. Why? When a thing does not need to be moved, and is not desirable to have a Qubes-rendered outline around it (as all menus have, today), it would be preferred to more elegantly be able to mimic overlays as Slack does, below. 

Specifically, the on-hover fly-outs for the Journalist Badge, the Logged In User's badge, and the Seen By bubbles invokable [in this prototype](https://www.figma.com/proto/vs52zFnp5zpEHfCnTI0R3U/SDW-Design-System?page-id=0%3A1&node-id=7%3A2636&viewport=241%2C48%2C0.5&scaling=min-zoom&starting-point-node-id=7%3A2636). 

<img width="1090" alt="Screen Shot 2021-12-08 at 12 06 30 AM" src="https://user-images.githubusercontent.com/8262612/145172401-220355b9-4b8d-4341-83c6-fe56e8dfd084.png">

![Menus-2ndrev](https://user-images.githubusercontent.com/8262612/145173125-19a2cf42-158c-47f2-b576-66868fb30a8a.png)

**Comment by eloquence** [comment]
I agree it's worth at least a first timeboxed investigation (in a future sprint) for us all to be able to better understand and reason about the interaction between GUI toolkit, window manager, and OS. This is important as we add more flyouts, custom tooltips, etc

My current understanding is this:

- Qubes adds colored decoration to anything that's (on a technical level) a separate window. That's anything that can _extend beyond the boundaries of the window you're in_. Note, for example, in Slack, none of the menus can ever extend beyond the window boundary. Slack is pretty smart about scaling them down as needed - making them scrollable.

- If you experiment in other apps in Qubes, you'll notice the decoration around pull-down menus, context menus, etc. It doesn't matter whether they're built with Qt, GTK, or something else. There are some exceptions -- try opening `gedit`, for example, and you'll notice that its menus don't have the Qubes decoration. They also cannot extend beyond the window boundary (they just get cut off).

In other words, I think we'll want to find out if there are good ways to render widgets on top of other widgets in a manner that's not technically a separate window -- but we will then also have to make sure that they're not cut off. This could be the kind of problem it could be useful to collaborate with Felgo on, once we've done our own investigation.

**Comment by eloquence** [comment]
Closing, non-critical issue specific to the current codebase/UX, which is undergoing a rewrite.
